local makefile changes

From: Pete Wyckoff <pw@osc.edu>

Build osdemu iscsi target only.  Link with local osd-target emulator code.
---

 usr/Makefile |   22 +++++++++++++++++++---
 1 files changed, 19 insertions(+), 3 deletions(-)


diff --git a/usr/Makefile b/usr/Makefile
index 5e92302..7b83d92 100644
--- a/usr/Makefile
+++ b/usr/Makefile
@@ -1,3 +1,8 @@
+# local hacks to avoid needing a higher-level Makefile
+ISCSI := 1
+OSDEMU := 1
+INCLUDES := -DHAVE_VALGRIND
+
 ifneq ($(IBMVIO),)
 CFLAGS += -DIBMVIO -DUSE_KERNEL
 TGTD_OBJS += $(addprefix ibmvio/, ibmvio.o)
@@ -43,11 +48,16 @@ bs_osdemu.o: bs_osdemu.c $(DEPLIBS)
 	$(CC) -c $(CFLAGS) -I$(OSD_TARGET_DIR) $< -o $@
 endif
 
-INCLUDES += -I. -I../include -I$(KERNELSRC)/include
+INCLUDES += -I. -I../include
 
 CFLAGS += -D_LARGEFILE64_SOURCE
 CFLAGS += $(INCLUDES)
-CFLAGS += -g -O2 -Wall -Wstrict-prototypes -fPIC
+CFLAGS += -g -Wall -Wstrict-prototypes -fPIC
+#CFLAGS += -Wcast-align -Wbad-function-cast -Wmissing-prototypes
+#CFLAGS += -Wundef -Wmissing-declarations -Wnested-externs
+#CFLAGS += -Wextra -Wno-sign-compare -Wno-unused -Wshadow
+#CFLAGS += -Wpointer-arith -Wcast-qual -Wwrite-strings
+
 
 LIBS += -lpthread
 
@@ -80,4 +90,10 @@ install: $(PROGRAMS)
 
 .PHONY: clean
 clean:
-	rm -f *.[od] $(PROGRAMS) iscsi/*.[od] ibmvio/*.[od] fc/*.[od]
+	rm -f *.[od] $(PROGRAMS) iscsi/*.[od] ibmvio/*.[od] fc/*.[od] tags
+
+# make tags
+.PHONY: tags
+tags:
+	ctags $$(find . -name '*.[ch]')
+
